warn-stranded.lua

```diff
@@ -0,0 +1,204 @@
+  end
+  return "("..sym..")"
+end
+
+local function getUnitDescription(unit)
+   return '['..dfhack.units.getProfessionName(unit)..'] '..dfhack.TranslateName(dfhack.units.getVisibleName(unit))..
+      ' '..getSexString(unit.sex)..' Stress category: '..dfhack.units.getStressCategory(unit)
+end
+
+
+local function unitIgnored(unit)
+   local currentIgnore = dfhack.persistent.get('warnStrandedIgnore')
+   if currentIgnore == nil  then return false end
+
+   for index, id in pairs(currentIgnore['ints']) do
+      if id == unit.id then
+         print(id)
+         print(unit.id)
+         print('Unit Ignored')
+         return true, index
+      end
+   end
+
+   return false
+end
+
+local function toggleUnitIgnore(unit)
+   local currentIgnore = dfhack.persistent.get('warnStrandedIgnore')
+
+   if currentIgnore == nil then
+      currentIgnore = { key = 'warnStrandedIgnore', value = 'warnStrandedIgnore', ints = {} }
+      dfhack.printerr('nil, creating')
+      table.insert(currentIgnore['ints'], unit.id)
+   end
+
+   local ignored, index = unitIgnored(unit)
+
+   if ignored then
+      dfhack.printerr('toggle remove')
+      table.remove(currentIgnore['ints'], unit.id)
+   else
+      dfhack.printerr('toggle insert')
+      table.insert(currentIgnore['ints'], unit.id)
+   end
+
+   dfhack.persistent.save(currentIgnore)
+end
+
+function warning:initListChoices(units)
+   local choices = {}
+   for _, unit in pairs(units) do
+      table.insert(choices, { text = getUnitDescription(unit), unit = unit })
+   end
+   local list = self.subviews.list
+   list:setChoices(choices, 1)
+end
+
+function warning:updateBottom()
+   self.subviews.bottom_ui:setText(
+      {
+         { key = 'SELECT', text = ': Toggle ignore unit', on_activate = self:callback('onIgnore') }, ' ',
+         { key = 'CUSTOM_SHIFT_I', text = ': Ignore all', on_activate = self:callback('onIgnoreAll') }, ' ',
+         { key = 'CUSTOM_SHIFT_C', text = ': Clear all ignored', on_activate = self:callback('onClear') },
+      }
+   )
+end
+
+function warning:onIgnore()
+   local index, choice = self.subviews.list:getSelected()
+   local unit = choice.unit
+
+   toggleUnitIgnore(unit)
+end
+
+function warning:onIgnoreAll()
+   local currentIgnore = dfhack.persistent.get('warnStrandedIgnore')
+   local choices = self.subviews.list.getChoices()
+
+   if currentIgnore == nil then
+      currentIgnore = { key = 'warnStrandedIgnore', value = 'warnStrandedIgnore', ints = {} }
+   end
+
+   for _, choice in choices do
+      if not unitIgnored(unit) then
+         table.insert(currentIgnore['ints'], unit.id)
+      end
+   end
+
+   dfhack.persistent.save(currentIgnore)
+end
+
+function warning:onClear()
+   clear()
+end
+
+function warning:onDismiss()
+    view = nil
+end
+
+function doCheck()
+  local grouped = {}
+  local citizens = dfhack.units.getCitizens()
+
+  -- Pathability group calculation is from gui/pathable
+  for _, unit in pairs(citizens) do
+    local target = unit.pos
```

Comment:
`unit.pos` might not be valid if the unit is in a cage or in some other esoteric circumstances. Use `dfhack.units.getPosition(unit)` instead